kong/runloop/handler.lua

```diff
@@ -1478,7 +1488,17 @@ return {
         end
 
         if enabled_headers[headers.VIA] then
-          header[headers.VIA] = SERVER_HEADER
+          -- Kong does not support injected directives like 'nginx_location_proxy_http_version',
+          -- so we skip checking them.
+          local proxy_http_version = ctx.proxy_http_version or
+                                     kong.configuration.proxy_http_version or
+                                     DEFAULT_PROXY_HTTP_VERSION
+
+          local kong_outbound_via = proxy_http_version .. " " .. SERVER_HEADER
+
+          local resp_via = ngx.resp.get_headers(MAX_HEADERS)[headers.VIA]
```

Comment:
`ngx.resp.get_headers` has the default max headers `100` limitation, so we need not to pass this parameter.

Reply:
Should we pass a higher value? I just worry the `Via` header is located at the very end of the response headers.

Reply:
we already have a option in kong.conf, it is `kong.configuration.lua_max_req_headers`, 100 by default.

Reply:
@chronolaw , probably `lua_max_resp_headers`:

```
#lua_max_resp_headers = 100      # Maximum number of response headers to parse by default.
                                 #
                                 # This argument can be set to an integer between 1 and 1000.
                                 #
                                 # When proxying, Kong returns all the response headers
                                 # and this setting does not have any effect. It is used
                                 # to limit Kong and its plugins from reading too many
                                 # response headers.
```

It is on response side.

Reply:
yes, that is it.

Reply:
Just updated the code.